lenient.js

```diff
@@ -59,10 +59,10 @@ var oMatch = {
 };
```

Comment:
Can you make these use `Map()` instead? Then we don't need all those ugly `({}.hasOwnProperty.call` calls.